docs/www/tls-kubernetes.md

```diff
@@ -0,0 +1,375 @@
+---
+title: TLS for Kubernetes
+order: 0
+---
+
+# TLS for Kubernetes
+
+Talking over the internet with no encryption is something unthinkable these days. With Redpanda you can help secure your data by enabling [Transport Layer Security (TLS)](https://en.wikipedia.org/wiki/Transport_Layer_Security) across your clusters, with just a few configurations. 
+
+If you haven’t, please check our guide about [Encryption, authorization & authentication](https://vectorized.io/docs/acls/).
+
+Here we will dive deeper into how we can configure TLS inside a Kubernetes cluster.
+
+The Custom Resource Definition (CRD) of a Redpanda cluster includes three APIs: Kafka, Admin and Pandaproxy APIs. Each API can support TLS for server authentication and mTLS (Mutual TLS) for client authentication.
+
+TLS allows us to establish a secure connection between a client and a broker in which communication is encrypted. 
+
+Mutual TLS (mTLS), or 2-way TLS, is an authentication method in which a client attempting to establish a connection with the server (broker) must present its certificate. This way the server can decide whether the client can be trusted.
+
+## Getting started
+
+To demonstrate how to get up and running with TLS in Kubernetes, we’ll use a sample `yaml` configuration file for a one-node cluster.
+
+To enable TLS we need to add a listener entry under the API fields in our yaml configuration file, and then specify the subfield `tls` where we can enable TLS.
+
+For example, check that `kafkaApi` now have a new flag that says `tls: enabled: true`.
+
+Save this file as `yaml`:
+
+```yaml
+apiVersion: redpanda.vectorized.io/v1alpha1
+kind: Cluster
+metadata:
+ name: cluster-sample-tls
+spec:
+ image: "vectorized/redpanda"
+ version: "latest"
+ replicas: 1
+ resources:
+   requests:
+     cpu: 1
+     memory: 1.2G
+   limits:
+     cpu: 1
+     memory: 1.2G
+ configuration:
+   rpcServer:
+     port: 33145
+   kafkaApi:
+    - port: 9092
+      tls:
+        enabled: true
+   pandaproxyApi:
+    - port: 8082
+      tls:
+        enabled: true
+   adminApi:
+   - port: 9644
+   developerMode: true
+```
+
+We’ll use the Kafka API to go through the TLS configuration, but you can use similar settings to set up TLS for the other APIs. For this guide, we assume external connectivity is disabled unless otherwise stated.
```

Comment:
For Kubernetes, external connectivity means that you can access a cluster from outside of the network. To enable it, we have to setup a configuration file, with this particular field enabled. 

There's more info here
https://vectorized.io/docs/kubernetes-external-connect/

https://github.com/vectorizedio/redpanda/blob/dev/src/go/k8s/config/samples/external_connectivity.yaml

Basically, to enable `external_connectiviy` we have to setup a flag to enable it

Reply:
@dimitriscruz is it possible to remove this phrase? Is there loss of context or can something malfunction if we have external_connectivity? 

```suggestion
We’ll use the Kafka API to go through the TLS configuration, but you can use similar settings to set up TLS for the other APIs.

```

Reply:
The exposed addresses are different when there's external connectivity. Would adding a link to the external connectivity doc be sufficient? Also we could point to the part of this doc that considers external connectivity.

Reply:
I think adding a link should be sufficient for now.

Reply:
```suggestion
We’ll use the Kafka API to go through the TLS configuration, but you can use similar settings to set up TLS for the other APIs. For this guide, we assume external connectivity is disabled unless otherwise stated. If you want to enable external connectivity, check [this guide](https://vectorized.io/docs/kubernetes-external-connect/) that we wrote about it. 
```